packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -366,6 +415,42 @@ export class PeerPool extends EventEmitter {
 		return true;
 	}
 
+	public addOutboundPeer(
+		peerId: string,
+		peerInfo: P2PDiscoveredPeerInfo,
+		socket: SCClientSocket,
+	): boolean {
+		const existingPeer = this.getPeer(peerId);
+
+		if (existingPeer) {
+			// Update the peerInfo from the latest inbound socket.
+			existingPeer.updatePeerInfo(peerInfo);
+			if (existingPeer.state.outbound === ConnectionState.DISCONNECTED) {
+				existingPeer.outboundSocket = socket;
+
+				return false;
+			}
+
+			return false;
+		}
+
+		const peerConfig = {
+			connectTimeout: this._peerPoolConfig.connectTimeout,
+			ackTimeout: this._peerPoolConfig.ackTimeout,
+		};
+		const peer = new Peer(peerInfo, peerConfig);
+		peer.createPeerFromOutboundConnection(socket);
```

Comment:
As mentioned in a previous comment. It would be safer to pass the outbound socket to the `Peer` constructor.